op.c

```diff
@@ -2469,8 +2469,8 @@ Perl_scalarvoid(pTHX_ OP *arg)
         }
         o = next_kid;
     }
-
-    return arg;
+    /* we should never get here */
```

Comment:
Given the code at the end of the loop I don't think it's possible for that code to be reached.

Does simply changing this to `NOT_REACHED;` prevent the warning?

Reply:
Not sure what you mean by "changing this". I tried adding a NOT_REACHED before the `return arg;` but it still threw warnings. The HPUX compiler doesn't seem to recognize various conventions that gcc does.

Reply:
Sorry, I meant `NOT_REACHED;` instead of the return.

Though I'm surprised we don't see warnings for the missing return with the change as it is.

Reply:
dmq: i think i prefer while(1) + NOT_REACHED.
TonyC: I'm fine with that